### PR TITLE
postgresql_privs: fix wrong codec specification

### PR DESCRIPTION
**Pull request by b6d**
Use psycopg2's mapping from PostgreSQL encoding names to Python codec names.

### CODE REVIEW

**Comment by mpdehaan** [comment]
Are these extensions always going to be installed?

How is this problem reproduced?

**Comment by b6d** [comment]
psycopg2.extensions is part of psycopg2 (not extending psycopg2 but extending the Python DB API), so they should be installed if psycopg2 is installed.

The Problem I'm trying to address is as follows:

The encoding of a psycopg2 connection defaults to the database's encoding.

Exception Messages come in this encoding, e.g. `relation »foo« does not exist`, including french quotation marks for psycopg2 packaged with Ubuntu 12.04 talking to PostgreSQL 8.4 (On Arch Linux, I get "normal" double quotes talking to Postgres 9.2).

Returning these messages via `fail_json` led to UnicodeErrors in earlier versions of the module, so messages are now reencoded (`message.decode(...).encode(...)`).

`psycopg2.connection.encoding` is a PostgreSQL encoding name (e.g. "SQL_ASCII"), but the "decode" part needs a codec specification Python knows (e.g. "ASCII"). `psycopg2.extensions.encodings` maps one to the other.

**tl;dr:** To reproduce the Problem with the current version of the module:
- Create a database with encoding "SQL_ASCII": `CREATE DATABASE my_db TEMPLATE template0 ENCODING 'SQL_ASCII';`
- Try to set a privilege on a non-existing database-object: `./test-module -m ../library/database/postgresql_privs -a "db=my_db type=table obj=nonexistent_table privs=all roles=postgres"`
- This leads to `LookupError: unknown encoding: SQLASCII` in the current version of the module.

**Comment by mpdehaan** [comment]
merged, thanks!
